modules/sizeMappingV2.js

```diff
@@ -0,0 +1,454 @@
+  getUniqueIdentifierStr,
+  logInfo,
+  logError,
+  logWarn,
+  isValidMediaTypes,
+  isArrayOfNums
+} from '../src/utils';
+import { processNativeAdUnitParams } from '../src/native';
+import { adunitCounter } from '../src/adUnits';
+import includes from 'core-js/library/fn/array/includes';
+import { getHook } from '../src/hook';
+import {
+  validateBannerMediaType,
+  validateVideoMediaType,
+  validateNativeMediaType,
+  validateSizes
+} from '../src/prebid';
+
+// Maps auctionId to a boolean value, value is set to true if Adunits are setup to use the new size mapping, else it's set to false.
+const _sizeMappingUsageMap = {};
+
+// returns "true" if atleast one of the adUnit in the adUnits array has declared a Ad Unit or(and) Bid level sizeConfig
+// returns "false" otherwise
+export function isUsingNewSizeMapping(adUnits) {
+  let isUsingSizeMappingBool = false;
+  adUnits.forEach(adUnit => {
+    if (adUnit.mediaTypes) {
+      // checks for the presence of sizeConfig property at the adUnit.mediaTypes object
+      Object.keys(adUnit.mediaTypes).forEach(mediaType => {
+        if (adUnit.mediaTypes[mediaType].sizeConfig) {
+          if (isUsingSizeMappingBool === false) {
+            isUsingSizeMappingBool = true;
+          }
+        }
+      });
+
+      // checks for the presence of sizeConfig property at the adUnit.bids[].bidder object
+      adUnit.bids.forEach(bidder => {
+        if (bidder.sizeConfig) {
+          if (isUsingSizeMappingBool === false) {
+            isUsingSizeMappingBool = true;
+          }
+        }
+      });
+    }
+  });
+  return isUsingSizeMappingBool;
+}
+
+// returns "adUnits" array which have passed sizeConfig validation checks in addition to mediaTypes checks
+// deletes properties from adUnit which fail validation.
+function checkAdUnitSetupHook(adUnits) {
+  return adUnits.filter(adUnit => {
+    const mediaTypes = adUnit.mediaTypes;
+    if (!mediaTypes || Object.keys(mediaTypes).length === 0) {
+      logError(`Detected adUnit.code '${adUnit.code}' did not have a 'mediaTypes' object defined. This is a required field for the auction, so this adUnit has been removed.`);
+      return false;
+    }
+
+    if (mediaTypes.banner) {
+      const banner = mediaTypes.banner;
+      if (banner.sizes) {
+        adUnit = validateBannerMediaType(adUnit);
+      } else if (banner.sizeConfig) {
+        if (Array.isArray(banner.sizeConfig)) {
+          banner.sizeConfig.forEach(config => {
+            // verify if all config objects include "minViewPort" and "sizes" property.
+            // if not, remove the mediaTypes.banner object
+            const keys = Object.keys(config);
+            if (!(includes(keys, 'minViewPort') && includes(keys, 'sizes'))) {
+              logError(`Ad Unit: ${adUnit.code}: mediaTypes.banner.sizeConfig is missing required property minViewPort or sizes or both. Removing the invalid mediaTypes.banner from Ad Unit.`);
+              return delete adUnit.mediaTypes.banner;
```

Comment:
Given this block of code exists in a `forEach`, if one of the configs were 'bad' - it would still keep looping through even though the parent `mediaTypes.banner` object should have been deleted.  If that's not expected, could you take another look at the logic here (and probably later in the function).

Reply:
@jsnellbaker Hi, I made the change.
I needed some way to display the log messages, and also delete the banner/video mediaType. I've made a change, let me know if that's fine.